extensions/amp-youtube/1.0/amp-youtube.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Youtube} from './youtube';
+
+/** @const {string} */
+const TAG = 'amp-youtube';
+
+class AmpYoutube extends PreactBaseElement {}
```

Comment:
Use `VideoBaseElement` for imperative API/actions. See `amp-video` or [`amp-video-iframe`](https://github.com/ampproject/amphtml/pull/31055/) for an example.

```suggestion
class AmpYoutube extends VideoBaseElement {}
```